File: matlab/bslcorr.m
% bslcorrmat
% corrects 129 channel ar mat 
% by subtracting mean of baseline in samplepoints

function [data] = bslcorr(inmat, bslvec);

if isempty (bslvec), 
    bslvec = 1:size(inmat,2); 
end

for chan = 1 : size(inmat, 1)
data(chan,:) = inmat(chan,:)-mean(inmat(chan,bslvec),2);
end

File: freqtag/bslcorr.py
import numpy as np


def bslcorr(inmat: np.ndarray, bslvec: np.ndarray = None) -> np.ndarray:
    """
    Baseline corrects EEG data by subtracting mean of baseline in sample points.

    Args:
      inmat:
        (m sensors, n sample points) array.
        EEG data to baseline correct.
      bslvec:
        (j sample points) array.
        Vector containing indices of the data to use as the baseline.
        Defaults to indices of the whole dataset, n sample points.

    Returns:
      (m sensors, n sample points) array:
        Baseline corrected data.
    """
    # TODO: Input validation.

    if bslvec is None:
        sample_points = inmat.shape[1]
        bslvec = np.arange(sample_points)
    meanvec = np.mean(inmat[:, tuple(bslvec)], axis=1)
    return inmat - meanvec[:, None]
